syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "contrib/coms/protos/finance.proto";
import "contrib/coms/protos/protocol.proto";

message CumulativeMetrics {
    google.protobuf.Timestamp period_open = 1;
    google.protobuf.Timestamp period_close = 2;
    float returns = 3;
    float pnl = 4;
    float cash_flow = 5; //capital_used
    float starting_exposure = 6;
    float ending_exposure = 7;
    float starting_value = 8;
    float ending_value = 9;
    float starting_cash = 10;
    float ending_cash = 11;
    int64 longs_count = 12;
    int64 shorts_count = 13;
    float long_value = 14;
    float short_value = 15;
    float long_exposure = 16;
    float short_exposure = 17;
    float gross_leverage = 18;
    float net_leverage = 19;
}

message PeriodCommonMetrics {
    repeated Order orders = 1;
    repeated Transaction transactions = 2;
    repeated Position positions = 3;
}

message CumulativeRiskMetrics {
    float algorithm_volatility = 1;
    float benchmark_period_return = 2;
    float benchmark_volatility = 3;
    float algorithm_period_return = 4;
    float alpha = 5;
    float beta = 6;
    float sharpe = 7;
    float sortino = 8;
    float max_drawdown = 9;
    float max_leverage = 10;
}